compiler/rustc_middle/src/ty/util.rs

```diff
@@ -828,6 +832,61 @@ impl<'tcx> ty::TyS<'tcx> {
         }
     }
 
+    pub fn gc_layout(&'tcx self, tcx: TyCtxt<'tcx>, param_env: ty::ParamEnv<'tcx>) -> (u64, u64) {
+        let layout = tcx.layout_of(param_env.and(self)).unwrap();
+        let w_size = Size::from_bytes(size_of::<usize>());
+        let w_align = Align::from_bytes(w_size.bytes()).unwrap();
+
+        if layout.align.abi.bytes() < (w_size.bytes())
+            || self.is_no_trace(tcx.at(DUMMY_SP), param_env)
+        {
+            // If an ADT is aligned to a size smaller than 1 word, then it
+            // contains no fields large enough to represent a GC pointer.
+            //
+            // By default, the collector will conservatively scan a struct's
```

Comment:
I don't understand this paragraph. I assume it's trying to explain the `|| self.is_no_trace(tcx.at(DUMMY_SP), param_env)` part of the conditional?

Reply:
Also, I *think* this optimisation is subsumed by the conditional at line 863 "If a field is smaller than a word"? Maybe we could just leave it to that point to make this optimisation?

Reply:
I've removed this comment (924a6889dd9734636f3ddbc43512fcdc24da09f3) because I think you're right it doesn't add much and is a bit confusing.